pkg/roachprod/logger/log.go

```diff
@@ -214,6 +214,13 @@ func (l *Logger) RootLogger() *Logger {
 // If the parent Logger was logging to a file, the new Logger will log to a file
 // in the same dir called <name>.log.
 func (l *Logger) ChildLogger(name string, opts ...loggerOption) (*Logger, error) {
+	// If this logger instance is redirecting to `/dev/null` (most
+	// likely tests), there's no point trying to create a child
+	// logger. Return the current logger instance itself.
+	if l.path == "/dev/null" {
```

Comment:
Nit: premature optimization? It looks like `/dev/null` is only passed in `planner_test.go`; there are more uses of `""` elsewhere. Add another disjunct or drop?

Reply:
Ah yes, I forgot how the logger handles empty string paths. That's better. Removed.